### PR TITLE
Protect form submit from pre-init use (close 4360)

### PR DESCRIPTION
**Pull request by bill-looby-i**
The details of the requirement and the occurrence are contained in Issue #4360. This fix has been tested against the issue noted there and fixes it.

Simple addition to prevent use of contextStorage if not initialised. This is a valid check as it can't be already sent (therefore the flag check isn't required) if the storage isn't even initialised.

Other holes may exist with the same pattern but only the form submit is handled here. The other likely candidate is `_onJsError()` if the error is triggered prior to the driver start.

### CODE REVIEW

File: src/client/driver/driver.js [hunk 1]
```diff
@@ -210,7 +210,7 @@ export default class Driver {
     // We have to cancel every form submit after a test is done
     // to prevent requests to a closed session
     _onFormSubmit (e) {
-        if (this.contextStorage.getItem(TEST_DONE_SENT_FLAG))
+        if (this.contextStorage !== null && this.contextStorage.getItem(TEST_DONE_SENT_FLAG))
```
Comment (AlexKamaev) [inline src/client/driver/driver.js]: I would prefer `if (this.contextStorage && this.contextStorage.getItem(...`, although your way is 100% correct in this case

Comment (AlexKamaev) [reply src/client/driver/driver.js]: @AndreyBelym take a look on this PR please.
We need to check why the 'contextStorage' in not defined, however I think this PR can be merged now

Comment (bill-looby-i) [reply src/client/driver/driver.js]: @AlexKamaev, @AndreyBelym See #4360, It's a race condition. In an IFrame, `driver.start()` is called on a promise, so a form that submits immediately triggers the `_onFormSubmit()` before `driver.start()` runs. For a main window this doesn't happen as `driver.start()` is called synchronously.

**Comment by AlexKamaev** [comment]
@testcafe-build-bot retest

**Comment by AlexKamaev** [comment]
@testcafe-build-bot retest

**Comment by AlexKamaev** [comment]
@AndreyBelym I'll add the test. I prepared example when researched the issue

**Comment by AlexKamaev** [comment]
@testcafe-build-bot retest

**Comment by AlexKamaev** [comment]
@testcafe-build-bot retest

**Pull request merged**